pycbc/inference/sampler_emcee.py

```diff
@@ -0,0 +1,994 @@
+# Copyright (C) 2016  Collin Capano
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""
+This modules provides classes and functions for using different sampler
+packages for parameter estimation.
+"""
+
+import numpy
+from pycbc.inference.sampler import _BaseMCMCSampler
+from pycbc.io import WaveformArray, FieldArray
+from pycbc.filter import autocorrelation
+
+#
+# =============================================================================
+#
+#                                   Samplers
+#
+# =============================================================================
+#
+
+
+class EmceeEnsembleSampler(_BaseMCMCSampler):
+    """This class is used to construct an MCMC sampler from the emcee
+    package's EnsembleSampler.
+
+    Parameters
+    ----------
+    likelihood_evaluator : likelihood class
+        An instance of the likelihood class from the
+        pycbc.inference.likelihood module.
+    nwalkers : int
+        Number of walkers to use in sampler.
+    processes : {None, int}
+        Number of processes to use with multiprocessing. If None, all available
+        cores are used.
+    burn_in_iterations : {None, int}
+        Set the number of burn in iterations to use. If None,
+        `burn_in_ieterations` will be initialized to 0.
+    """
+    name = "emcee"
+
+    def __init__(self, likelihood_evaluator, nwalkers, processes=None,
+                 burn_in_iterations=None):
+
+        try:
+            import emcee
+        except ImportError:
+            raise ImportError("emcee is not installed.")
+
+        # initialize the pool to use
+        if processes == 1:
+            pool = None
+        else:
+            pool = emcee.interruptible_pool.InterruptiblePool(
+                processes=processes)
+
+        # construct the sampler
+        ndim = len(likelihood_evaluator.waveform_generator.variable_args)
+        sampler = emcee.EnsembleSampler(nwalkers, ndim, likelihood_evaluator,
+            pool=pool)
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->